internal/replication/blockreplicator.go

```diff
@@ -355,6 +365,8 @@ func (br *BlockReplicator) processLeaderChanges(leader uint64) {
 				br.lg.Panicf("Error computing base header hash of last commited block: %+v; error: %s",
 					br.lastCommittedBlock.GetHeader(), err)
 			}
+			br.numInFlightBlocks = 0
```

Comment:
why no lock for this? The same is applicable for the other locations too. Use atomic type instead.

Reply:
This is in a method that starts with 
```
func (br *BlockReplicator) processLeaderChanges(leader uint64) {
	br.mutex.Lock()
	defer br.mutex.Unlock()
```

Reply:
... and I did forget to remove the TODO in line 345 ;-) which I'll fix